drivers/dma/dma_mcux_lpc.c

```diff
@@ -188,10 +194,66 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 
 	LOG_DBG("channel is %d", p_handle->channel);
 
-	if (config->source_chaining_en && config->dest_chaining_en) {
-		LOG_DBG("link dma out 0 to channel %d", config->linked_channel);
-		/* Link DMA_OTRIG 0 to channel */
-		INPUTMUX_AttachSignal(INPUTMUX, 0, config->linked_channel);
+	if (!config->source_chaining_en && !config->dest_chaining_en) {
```

Comment:
usually we get channel by dma_request_channel, and in this function, we actually has a atomic bit selection for channels, and when we configure the channle actually the channel shall be filterred already. so I suggest to add a new api like dma_request_linked_channel to check whether the channel is ready to link.